jormungandr/src/fragment/pool.rs

```diff
@@ -305,6 +307,18 @@ impl Pools {
             block_date.into(),
         );
     }
+
+    async fn get_current_block_date(&self) -> BlockDate {
+        let time = std::time::SystemTime::now();
+        let tip = self.tip.get_ref().await;
```

Comment:
No way we can just use `tip.block_date() + 1` (with some complication as needed to regard for eras)?

Reply:
But what about possible missed slots?

Reply:
Right, we can't use that.